### PR TITLE
Dks keyboards

### PR DESCRIPTION
**Pull request by itskevin-zz**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/ndk/ndk16/config.h
Comment (waffle87) [inline keyboards/handwired/ndk/ndk16/config.h:1]: This file should be removed.

File: keyboards/handwired/ndk/ndk16/info.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{
+    "manufacturer": "ndk",
+    "keyboard_name": "ndk16",
+    "maintainer": "itskevin-zz",
+    "bootloader": "caterina",
```
Comment (waffle87) [inline keyboards/handwired/ndk/ndk16/info.json:5]:
```suggestion
    "development_board": "promicro",
```

File: keyboards/handwired/ndk/ndk16/info.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    "build": {
+        "lto": true
+    },
+    "diode_direction": "COL2ROW",
+    "encoder": {
+        "rotary": [
+            {"pin_a": "F5", "pin_b": "F4", "resolution": 4}
```
Comment (waffle87) [inline keyboards/handwired/ndk/ndk16/info.json:12]:
```suggestion
            {"pin_a": "F5", "pin_b": "F4"}
```

File: keyboards/handwired/ndk/ndk16/info.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "oled": true
+    },
+    "matrix_pins": {
+        "cols": ["D7", "D4", "C6", "B4", "E6"],
+        "rows": ["B6", "B2", "B3", "B1", "F7"]
+    },
+    "processor": "atmega32u4",
```
Comment (waffle87) [inline keyboards/handwired/ndk/ndk16/info.json:29]:
```suggestion
```

File: keyboards/handwired/ndk/ndk16/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+    oled_set_cursor(0,11);
+    if (oled_tracker) {
+        render_cat();
+    } else {
+        render_cat_yawn();
+    }
+
+    return false;
+}
+#endif
```
Comment (waffle87) [inline keyboards/handwired/ndk/ndk16/keymaps/default/keymap.c:254]: This should be moved to the keyboard level in an `ndk16.c` file using the appropriate `_kb` functions, as OLEDs are enabled at the keyboard level.
```suggestion
```

File: keyboards/handwired/ndk/ndk16/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+    oled_set_cursor(0,11);
+    if (oled_tracker) {
+        render_cat();
+    } else {
+        render_cat_yawn();
+    }
+
+    return false;
+}
+#endif
```
Comment (waffle87) [inline keyboards/handwired/ndk/ndk16/keymaps/via/keymap.c:254]: As above.
```suggestion
```

File: keyboards/handwired/ndk/ndk16/keymaps/via/rules.mk [hunk 6]
```diff
@@ -0,0 +1,4 @@
+VIA_ENABLE = yes
+ENCODER_ENABLE = yes
+ENCODER_MAP_ENABLE = yes
+OLED_ENABLE = yes
```
Comment (waffle87) [inline keyboards/handwired/ndk/ndk16/keymaps/via/rules.mk:4]:
```suggestion
VIA_ENABLE = yes
ENCODER_MAP_ENABLE = yes
```

File: keyboards/handwired/ndk/ndk16/readme.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
+# ndk16
+
+![ndk16](https://github.com/itskevin-zz/ndk-keyboards/assets/7293885/739dfeeb-6cf0-4c67-9100-5f9a37c662bf)
```
Comment (waffle87) [inline keyboards/handwired/ndk/ndk16/readme.md:3]: This image needs to be rehosted on Imgur.

File: keyboards/handwired/ndk/ndk16/readme.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+The cat oled image [source](https://github.com/codeyfrommars/Gamepad).
+
+Make example for this keyboard (after setting up your build environment):
+
+    make handwired/nk16o:default
+
+Flashing example for this keyboard:
+
+    make handwired/nk16o:default:flash
```
Comment (waffle87) [inline keyboards/handwired/ndk/ndk16/readme.md:19]:
```suggestion
    make handwired/ndk/ndk16:default

Flashing example for this keyboard:

    make handwired/ndk/ndk16:default:flash
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/handwired/ndk/ndk16/readme.md (hunk 7 above)
Comment (itskevin-zz) [reply keyboards/handwired/ndk/ndk16/readme.md]: Changed url to Imgur

File: keyboards/handwired/ndk/ndk16/keymaps/default/keymap.c (hunk 4 above)
Comment (itskevin-zz) [reply keyboards/handwired/ndk/ndk16/keymaps/default/keymap.c]: Added ndk16.c, moved the OLED code there and changed functions to _kb.

Thanks

File: keyboards/handwired/ndk/ndk16/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+     * │   ├───┼───┼───┼───┤
+     * │   │ 1 │ 2 │ 3 │   │
+     * ├───┼───┴───┼───┤ ← │
+     * │ D │   0   │ . │   │
+     * └───┴───────┴───┴───┘
+     */
+    [0] = LAYOUT_numpad_5x5(
+        KC_MUTE, KC_NUM, KC_PSLS, KC_PAST,  KC_PMNS,
+        KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+        KC_P4,   KC_P5,   KC_P6,
```
Comment (drashna) [inline keyboards/handwired/ndk/ndk16/keymaps/default/keymap.c:37]: This should follow the other numpad layouts.  Eg: 

```suggestion
        KC_P7,   KC_P8,   KC_P9,
        KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
```

File: keyboards/handwired/ndk/ndk16/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+        KC_P1,   KC_P2,   KC_P3, 
+        TO(1), KC_P0,   KC_PDOT,  KC_PENT
+    ),
+    [1] = LAYOUT_numpad_5x5(
+        _______,   _______,   _______,   _______,   _______,
+        _______,   _______,   _______,   _______,
+        _______,   C(KC_A),   _______,
```
Comment (drashna) [inline keyboards/handwired/ndk/ndk16/keymaps/default/keymap.c:44]: And
```suggestion
        _______,   _______,   _______,
        _______,   C(KC_A),   _______,   _______,
```

etc

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/ndk/ndk16/keymaps/default/keymap.c (hunk 9 above)
Comment (itskevin-zz) [reply keyboards/handwired/ndk/ndk16/keymaps/default/keymap.c]: updated the matrix and numpad layout.

File: keyboards/handwired/ndk/ndk16/keymaps/default/keymap.c (hunk 10 above)
Comment (itskevin-zz) [reply keyboards/handwired/ndk/ndk16/keymaps/default/keymap.c]: see above

File: keyboards/handwired/ndk/ndk16/ndk16.c [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+
+#if defined(OLED_ENABLE)
+
+/* oled fix for 'cold boot' */
+void keyboard_post_init_user(void) {
+    wait_ms(2000);
+    oled_init(OLED_ROTATION_0);
+}
```
Comment (drashna) [inline keyboards/handwired/ndk/ndk16/ndk16.c:145]: This shouldn't be needed, and is odd that it is.

Also, function should be the `kb` version, not the user.

File: keyboards/handwired/ndk/ndk16/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /*
+     * ┌───┬───┬───┬───┬───┐
```
Comment (drashna) [inline keyboards/handwired/ndk/ndk16/keymaps/default/keymap.c:22]: Hmm, this doesn't match the numpad layout style. numapd_5x5 should look like this: 
```c
    /*
     * ┌───┌───┬───┬───┬───┐
     * │ , │Num│ / │ * │ - │
     * ├───┼───┼───┼───┼───┤
     * │Gui│ 7 │ 8 │ 9 │   │
     * ├───┼───┼───┼───┤ + │
     * │Alt│ 4 │ 5 │ 6 │   │
     * ├───┼───┼───┼───┼───┤
     * │Shi│ 1 │ 2 │ 3 │   │
     * ├───┼───┴───┼───┤Ent│
     * │Ctr│ 0     │ . │   │
     * └───┴───────┴───┴───┘
     */
```

The layout name should remove the 5x5 part.

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**